opm/simulators/utils/readDeck.cpp

```diff
@@ -545,6 +545,16 @@ void Opm::readDeck(Opm::Parallel::Communication    comm,
                    const bool                      checkDeck,
                    const std::optional<int>&       outputInterval)
 {
+
+    if (comm.rank() == 0) {
+        std::shared_ptr<Opm::StreamLog> streamLog = std::make_shared<Opm::StreamLog>(std::cout, Opm::Log::StdoutMessageTypes);
+        Opm::OpmLog::addBackend("STDOUT_LOGGER", streamLog);
+        // Set a tag limit of 10 (no category limit).
+        streamLog->setMessageLimiter(std::make_shared<Opm::MessageLimiter>(10));
+        const bool use_color_coding = OpmLog::stdoutIsTerminal();
+        streamLog->setMessageFormatter(std::make_shared<Opm::SimpleMessageFormatter>(use_color_coding));
+    }
```

Comment:
At least create a helper function for this part so we don't have to duplicate the implementation here and in the "official" setup function.

Reply:
On second thought, we should not include defining the logging system as part of reading the input.  Please pull this out of `readDeck()`.

Reply:
I do not know the history, but in flow, the logging system is setup up in the Main::readDeck() function. But I agree with you, it should be outside of it. Will update.  Just not very familiar with this part of the code. 

```c++
void Main::readDeck(const std::string& deckFilename,
                    const std::string& outputDir,
                    const std::string& outputMode,
                    const bool init_from_restart_file,
                    const bool allRanksDbgPrtLog,
                    const std::string& parsingStrictness,
                    const std::string& actionParsingStrictness,
                    const std::string& inputSkipMode,
                    const std::size_t numThreads,
                    const int output_param,
                    const std::string& parameters,
                    std::string_view moduleVersion,
                    std::string_view compileTimestamp)
{
    auto omode = setupLogging(FlowGenericVanguard::comm(),
                              deckFilename,
                              outputDir,
                              outputMode,
                              outputCout_, "STDOUT_LOGGER", allRanksDbgPrtLog);
```